### PR TITLE
add delete_file function and get_bucket_id

### PR DESCRIPTION
**Pull request by marcorosa**

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
Hell yes!

I haven't pulled this yet but wanted to first express my excitement and gratitude to have some help with this.

I'm almost done troubleshooting travis integration - do you think you could write some tests to go along with this (and maybe by then we can have them running in CI)?

**Comment by marcorosa** [comment]
I was considering whether to write some tests, but I think this should be done only once file upload and download functions are implemented.
Otherwise, we could only test file deletion with files uploaded with the c-libstorj, but uploading files with the c-libstorj before performing python unit tests would make actual them fail (such as issue #3 on your branch, my last post)

**Comment by bryanchriswhite** [comment]
That's fair. If you feel like messing with it though, you could try do do it progromatically like in [`TestListFileSuccess#setUp`:242-245](https://github.com/Storj/python-libstorj/blob/master/tests/test_storj_env.py#L245); but that's the part that's been causing issues lately, so it may be not worth fixing because `store_file` is the next thing on the todo.

File: lib/ext/python_libstorj.cpp [hunk 1]
```diff
@@ -64,6 +64,37 @@ void delete_bucket_cb(uv_work_t *work_req, int status) {
...
+    char *error_str = (char *)calloc(255, sizeof(char));
+    PyObject *error = Py_None;
+    get_bucket_id_request_t *req = (get_bucket_id_request_t *)work_req->data;
+    PyObject *_handle = (PyObject *)req->handle;
+
+    PyObject *bucket_dict = PyDict_New();
+    if (error_and_status_check<get_bucket_id_request_t>(req, &error_str)) {
+        PyDict_SetItemString(bucket_dict, "name", PyString_FromString(req->bucket_name));
+        PyDict_SetItemString(bucket_dict, "id", PyString_FromString(req->bucket_id));
+        // TODO: manage encrypted bucket name
```
Comment (bryanchriswhite) [inline lib/ext/python_libstorj.cpp]: @marcorosa what did you have in mind here?

For reference:
- [storj.c:184](https://github.com/Storj/libstorj/blob/893a729579c8e68afd844d2fc949a4e786922d87/src/storj.c#L184) is where libstorj is encrypting the bucket name (before making a request to the [`/bucket-ids/` route of bridge](https://github.com/Storj/bridge/blob/master/lib/server/routes/buckets.js#L123) at [storj.c:203](https://github.com/Storj/libstorj/blob/893a729579c8e68afd844d2fc949a4e786922d87/src/storj.c#L203) using the path from [storj.c:197](https://github.com/Storj/libstorj/blob/893a729579c8e68afd844d2fc949a4e786922d87/src/storj.c#L197)).
- [crypto.c:185](https://github.com/Storj/libstorj/blob/23d5c55e7e0a0564dcd4c9a9199bf0612279f025/src/crypto.c#L185) is where `encrypt_bucket_name` is defined.

**Pull request merged**